luigi/contrib/dropbox.py

```diff
@@ -0,0 +1,275 @@
+        dirs = []
+        lister = self.conn.files_list_folder(path, recursive=True, **kwargs)
+        dirs.extend(lister.entries)
+        while lister.has_more:
+            lister = self.conn.files_list_folder_continue(lister.cursor)
+            dirs.extend(lister.entries)
+        return [d.path_display for d in dirs]
+
+    def move(self, path, dest):
+        self.conn.files_move_v2(from_path=path, to_path=dest)
+
+    def copy(self, path, dest):
+        self.conn.files_copy_v2(from_path=path, to_path=dest)
+
+    def download_as_bytes(self, path):
+        metadata, response = self.conn.files_download(path)
+        return response.content
+
+    def upload(self, tmp_path, dest_path):
+        with open(tmp_path, 'rb') as f:
+            file_size = os.path.getsize(tmp_path)
+
+            CHUNK_SIZE = 4 * 1000 * 1000
+            upload_session_start_result = self.conn.files_upload_session_start(f.read(CHUNK_SIZE))
+            commit = dropbox.files.CommitInfo(path=dest_path)
+            cursor = dropbox.files.UploadSessionCursor(session_id=upload_session_start_result.session_id,
+                                                       offset=f.tell())
+
+            if f.tell() >= file_size:
+                self.conn.files_upload_session_finish(f.read(CHUNK_SIZE), cursor, commit)
+                return
+
+            while f.tell() < file_size:
+                if (file_size - f.tell()) <= CHUNK_SIZE:
+                    self.conn.files_upload_session_finish(f.read(CHUNK_SIZE), cursor, commit)
+                else:
+                    self.conn.files_upload_session_append_v2(f.read(CHUNK_SIZE), cursor)
+                    cursor.offset = f.tell()
+
+
+class ReadableDropboxFile(object):
+    def __init__(self, path, client):
+        """
+        Represents a file inside the Dropbox cloud which will be read
+
+        :param str path: Dropbpx path of the file to be read (always starting with /)
+        :param DropboxClient client: a DropboxClient object (initialized with a valid token)
+
+        """
+        self.path = path
+        self.client = client
+        self.download_file_location = os.path.join(tempfile.mkdtemp(prefix=str(datetime.datetime.utcnow())),
+                                                   ntpath.basename(path))
+        self.fid = None
+        self.closed = False
+
+    def read(self):
+        return self.client.download_as_bytes(self.path)
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc, traceback):
+        self.close()
+
+    def __del__(self):
+        self.close()
+        if os.path.exists(self.download_file_location):
+            os.remove(self.download_file_location)
+
+    def close(self):
+        self.closed = True
+
+    def readable(self):
+        return True
+
+    def writable(self):
+        return False
+
+    def seekable(self):
+        return False
+
+    def seek(self, offset, whence=None):
+        pass
```

Comment:
why include this and not do anything?

Reply:
You are right, I have just removed it.

I saw in other targets (e.g. AzureBlob, GCS) that their `ReadableFile` classes has a seek method that did nothing, so I assumed it was a required method. Reading the code for S3Target, I see that, indeed, it is not required.

Reply:
Were those other examples of readable files inheriting from a common source which included seek? yours only inherits from `object`